### PR TITLE
Add Nextflow Support

### PR DESCRIPTION
**Pull request by edmundmiller**
Adding support for the new Nextflow Language Server.

https://seqera.io/blog/modernizing-nextflow-developer-experience/

Would love some feedback on anything to clean up. It also hangs whenever I try to download the lsp with `lsp-install-server`

### CODE REVIEW

**Comment by jcs090218** [comment]
Can you update our CHANGELOG? Thanks! :D

**Comment by edmundmiller** [comment]
Done! Thanks @jcs090218!

File: clients/lsp-nextflow.el [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+(lsp-defcustom lsp-nextflow-debug nil
+  "Enable debug logging and debug information in hover hints."
+  :type 'boolean
+  :group 'lsp-nextflow
+  :package-version '(lsp-mode . "9.0.0")
+  :lsp-path "nextflow.debug")
+
+(lsp-defcustom lsp-nextflow-files-exclude [".git" ".nf-test" "work"]
+  "Configure glob patterns for excluding folders from being searched for Nextflow scripts and configuration files."
```
Comment (Lenbok) [inline clients/lsp-nextflow.el:85]: Compile warning, docstring wider than 80 chars.

File: clients/lsp-nextflow.el [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  "Use the [Harshil Alignment™️](https://nf-co.re/docs/contributing/code_editors_and_styling/harshil_alignment) when formatting Nextflow scripts and config files.
+
+  *Note: not all rules are supported yet*"
+  :type 'boolean
+  :group 'lsp-nextflow
+  :package-version '(lsp-mode . "9.0.0")
+  :lsp-path "nextflow.formatting.harshilAlignment")
+
+(lsp-defcustom lsp-nextflow-java-home nil
+  "Specifies the folder path to the JDK. Use this setting if the extension cannot find Java automatically."
```
Comment (Lenbok) [inline clients/lsp-nextflow.el:101]: Compile warning, docstring wider than 80 chars.

**Comment by jcs090218** [comment]
Can you resolve the compile warnings? Thanks! :)

```
In toplevel form:
lsp-nextflow.el:84:2: Warning: custom-declare-variable `lsp-nextflow-files-exclude' docstring wider than 80 characters
lsp-nextflow.el:100:2: Warning: custom-declare-variable `lsp-nextflow-java-home' docstring wider than 80 characters
```

**Comment by edmundmiller** [comment]
@jcs090218 Done, thank you!

File: clients/lsp-nextflow.el [hunk 3]
```diff
@@ -0,0 +1,140 @@
+;;; .local/straight/repos/lsp-mode/clients/lsp-nextflow.el -*- lexical-binding: t; -*-
+
+
```
[the remark this answers is not in the document]
Comment (edmundmiller) [reply clients/lsp-nextflow.el:2]: Good catch, updated the file name as well!

**Review by jcs090218 — APPROVED** [review]

**Comment by Lenbok** [comment]
It would be nice to get the server install working. I had to point the client at the jarfile manually downloaded. Surely since the jarfile is just a download from a GitHub release that should be straight forward? @jcs090218 ?

**Comment by edmundmiller** [comment]
> It would be nice to get the server install working. I had to point the client at the jarfile manually downloaded. Surely since the jarfile is just a download from a GitHub release that should be straight forward? @jcs090218 ?

I agree, but I think we can address that in a follow-up PR unless you have time to hack that out. I probably won't have time and would like to see this get merged before it gets stale.

**Comment by jcs090218** [comment]
> It would be nice to get the server install working. I had to point the client at the jarfile manually downloaded. Surely since the jarfile is just a download from a GitHub release that should be straight forward? @jcs090218 ?

I agree. It would be nice to have the server installation working, and it's indeed straightforward since I've done it a couple of times myself. :)

> I agree, but I think we can address that in a follow-up PR unless you have time to hack that out. I probably won't have time and would like to see this get merged before it gets stale.

No problem! I would like to see smaller PRs since it's usually easier to build progress that way. 🤔

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Merged! Thank you for your contribution! :D

**Comment by Bio-finder** [comment]
Hello, 
I'm an emacs user of lsp-mode. How is it possible to install and run this server with nf files? I've looked on the internet but didn't find the answer anywhere.

**Comment by jcs090218** [comment]
Maybe open a discussion for the record. 🤔

You can check the language manual from our website https://emacs-lsp.github.io/lsp-mode/page/lsp-nextflow/.

**Comment by Bio-finder** [comment]
Sorry, 
I started to look into the code of the lsp-mode and found the variable that I needed to link the lsp server to lsp so it could be loaded :) . Thank you for coding such an amazing tool!

**Pull request merged**